### ISSUE TITLE
RFC: Better e2e testing

### ISSUE THREAD

**Issue by BackEndTea** [opencfp/opencfp#774]
Right now we use our controller integration tests as e2e tests, but they widely differ in what they actually test. Some mock out just about everything and try to be it a unit test, while others mock next to nothing.

We also run into the issue that while PHPUnit is great for unit tests, its maybe not the best choice for e2e testing. I think we may greatly benefit from something like [behat](http://behat.org/en/latest/) to do these kind of big tests.

**Comment by BackEndTea** [comment]
Some options that i found

* [Codeception](http://codeception.com/docs/03-AcceptanceTests)
* [Behat](http://behat.org/en/latest/)
* [StoryPlayer](http://datasift.github.io/storyplayer/)

**Comment by mdwheele** [comment]
We used to have a Behat suite (that I used originally to give me some stability while making large changes). I would prefer it to Codeception.

TBH, I think we go test crazy on controllers and a good, solid and readable e2e test would better express the behaviours we're after. The fact that we have like 90km of mock for 1m of test is a signal I think.

I would personally be very-much behind a Behat-based e2e suite as long as it described *behaviour*, not over-specification / over-use of "browser-based clicky clicky". IMO, that "clicky clicky" belongs in the Context, while Gherkin should clearly describe behaviour that happens to implemented via web interface (or an `ApiContext`). Same behaviour (Gherkin) in multiple contexts (`WebContext`, `ApiContext`)
